### PR TITLE
Restructure ZenBytes

### PR DESCRIPTION
**Pull request by fa9r**
I restructured/revamped ZenBytes: 
- Instead of 3 large notebooks, we now have 7 smaller lessons (more coming soon) that are centred around specific MLOps topics.
- The README has been redesigned from ground up to better communicate what exactly ZenBytes is and how it relates to ZenML.
- The old notebook "00 - Basics" was completely removed as the content was not related to ML (and too trivial IMO).
- The content of the 7 new ZenByte lessons is largely taken from the old ZenBytes 01 and 02, but has been majorly rearranged, commented, and generally polished.
- Newly added content: experiment tracking with W&B
- Added 'Open in Colab' buttons (notebooks not tested in Colab yet)

Next Steps:
- add overview of entire MLOps stack being formed
- add more visualizations
- chapter 3: add drift detection between training and serving in 3.1
- chapter 3: add lesson on drift detection
- chapter 3: add lesson on exploratory data analysis
- chapter 3: add lesson on data validation
- chapter 4: split 4.1 into several smaller lessons to set up infrastructure, software, and MLOps stack one by one

### CODE REVIEW

**Comment by htahir1** [comment]
Wow what an amazing job! I really like the new flow and what we're going for now. Very neat and unified. I have some minor comments here and there which ill document:

# General
* I think the `1-1` , `2-1` convention is a bit hard to follow. Is there any reason for it other than legacy? Can we group them differently? And give the groups better names? Maybe something like madewithml loosely:

image

* Can we create profiles for each group for the `ZenML Setup` phase? I feel like it isolates work better and enforces best practices

* I feel like we're missing a big overview of the entire stack being formed. I imagine something where we are building a MLOps infrastructure stack step by step here. I believe @strickvl is already working on something similar for the docs and its a central piece of our messaging, so we can reuse it here probably?

* One chapter that would do well here is the visualization of statistics using Facets or whylogs. That might be a seperate chapter altogether called "Exploratory Data Analysis".  We can call this 3.1, and have drift detection as 3.2. We might also couple that with data validation (3.3) eventually when we have it. 

# Chapter by chapter

## 1.1 
* An **[ML Pipeline] -> A **[ML Pipeline] (I think(
* At the end, when we say "In the next lesson".. can we simply have a link that opens up the next notebook? Can we do this everytime it ends?

## 1.2
* Link dash
* Alternative to the paragraph about metadata stores: In addition to the artifact itself, ZenML automatically stores metadata [Metadata](https://docs.zenml.io/core-concepts#metadata-store), **e.g. where these artifacts are stored**, in a [Metadata Store](https://docs.zenml.io/core-concepts#metadata-store). By default, this a **SQLite** database on your local machine, but we could again easily switch this out for a cloud service.
* As a general comment, how about also adding the concept of an `orchestrator` here? It would complete the backbone of a default stack, i.e., metadata store, artifact store, and orchestrator

## 2.1
* In the intro, I would add some short explaination that we want the experiment tracker to be modular and abstracted away as much as possible from the pipeline. This would motivate a bit more the value of ZenML
* mlflow.sklearn.autolog() should be explained or linked
* For `wandb` I think we can use the PROJECT_NAME to construct a URL already: `https://wandb.ai/{WANDB_ENTITY}/{WANDB_PROJECT}/runs/`

## 2.2 
* This chapter could use a visualization for the service (maybe a subset of the GIF in the next chapter)
* Need to do `zenml stack set default` in this chapter because if you come from the previous chapter you're still in the wandb stack

## 2.3
* Nothing here to edit, maybe just a bit more explanation of how the two pipelines interact with each other and how they can run on different schedules

## 3.1
* I would rename this chapter to just Drift Detection and not use Evidently as the main thing here. Evidently is just a means to show Drift Detection
* We're missing a `!` in the first blog `zenml integration install evidently -f`
* I guess the Discord URL could be a variable. Im leaning towards getting rid of it altogether. Its too much setup
* This `(_assets/evidently+discord+mlflow.png)` image and subsequent images could be replaced by my suggestion in the #General feedbacl
* Might make sense to mention whylogs here as you said, but only  briefly. Its not really a drift detector.
* I feel like this chapter needs a bit more explanation about what the visualization is and how to actually understand the ramifications of it. We might want to flesh it out more with more sorts of drift, or explain the report and how to use it

## 4.1
* I guess this is the trickiest bit of everything. We might want to seperate in into chapters 4.1 metadata-store and artifact-store on s3, 4.2 container registry + Kubeflow local b)  c) secret manager and seldon +  Kubeflow AWS deployment 

That it for now. Hope its helpful! Thanks for the great work, ill boil this down into a quickstart now!

File: 4-1_Cloud_Deployment.ipynb [hunk 1]
```diff
@@ -0,0 +1,392 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Lesson 4.1: Deployment in the Cloud with Seldon & Kubeflow on AWS\n",
+    "\n",
+    "[![Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/4-1_Cloud_Deployment.ipynb)\n",
```
Comment (AlexejPenner) [inline 4-1_Cloud_Deployment.ipynb:9]: Not sure this one will run on colab, did you test it? The files will most probably not be present there?

File: README.md [hunk 2]
```diff
@@ -1,105 +1,119 @@
...
+|------------------------|-------------------------|---------------------|------------------------|
+| [1.1 ML Pipelines](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/1-1_Pipelines.ipynb)       | [2.1 Experiment Tracking](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/2-1_Experiment_Tracking.ipynb) | [3.1 Data Skew](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/3-1_Data_Skew.ipynb) | [4.1 Cloud Deployment](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/4-1_Cloud_Deployment.ipynb)   |
+| [1.2 Artifact Lifecycle](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/1-2_Artifact_Lineage.ipynb) | [2.2 Local Deployment](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/2-2_Local_Deployment.ipynb)    |                     |                        |
+|                        | [2.3 Inference Pipelines](https://colab.research.google.com/github/zenml-io/zenbytes/blob/main/2-3_Inference_Pipelines.ipynb) |                     |                        |
```
Comment (AlexejPenner) [inline README.md:26]: I love how this one looks

**Review by AlexejPenner — APPROVED** [review]
I really like the direction that this is going in!!! - besides the colab question I think this is good to go :)

File: 4-1_Cloud_Deployment.ipynb (hunk 1 above)
Comment (fa9r) [reply 4-1_Cloud_Deployment.ipynb:9]: For this specific notebook I am not sure if we can get it to work on Colab at all tbh. For the code, we could probably get it onto Colab with git clone, but you also need stuff like AWS CLI, Docker, Kubeflow, ... installed

Comment (AlexejPenner) [reply 4-1_Cloud_Deployment.ipynb:9]: Glad to have kicked off all these thoughts, looking forward to the changes :)

**Comment by strickvl** [comment]
Btw, not sure if you've heard of it, but https://www.reviewnb.com/ is a great tool for doing PR reviews on notebook files in case this becomes more of a thing?

**Comment by ayush714** [comment]
> Btw, not sure if you've heard of it, but https://www.reviewnb.com/ is a great tool for doing PR reviews on notebook files in case this becomes more of a thing?

Thanks Alex, it's helpful :)

**Comment by fa9r** [comment]
> Btw, not sure if you've heard of it, but https://www.reviewnb.com/ is a great tool for doing PR reviews on notebook files in case this becomes more of a thing?

Looks really cool, I'll add it to ZenBytes :)
